### PR TITLE
linter: Add lintignore feature and bypass mechanism

### PR DESCRIPTION
**Pull request by avisiedo**
Add support for filter files in the repository to avoid linting
them, and add a bypass mechanism which disable the filter, allowing
to filter files that was being filtered normally.
The bypass mechanism is enabled by setting LINT_FILTER_BYPASS=1

### CODE REVIEW

File: devel/lint.sh [hunk 1]
```diff
@@ -214,11 +219,20 @@ EOF
 }
 
 
+function is-in-lintignore
+{
+    local filepath="$1"
+    [ "${LINT_FILTER_BYPASS}" -eq 0 ] && grep -q "^${filepath}\$" devel/lint.ignore && return 0
+    return 1
```
Comment (frasertweedale) [inline devel/lint.sh]: One problem: `$filepath` has a leading `./`, e.g. `./devel/lint.sh`.  But the values in `devel/lint.ignore` are not expected to have the leading `./`.

Solution: strip the leading `./` from `$filepath` variable.  The following change is not perfect but perhaps good enough:
```diff
diff --git a/devel/lint.sh b/devel/lint.sh
index f39fa9c..7ca9876 100755
--- a/devel/lint.sh
+++ b/devel/lint.sh
@@ -221,7 +221,7 @@ EOF
 
 function is-in-lintignore
 {
-    local filepath="$1"
+    local filepath=${1/.\//}
     [ "${LINT_FILTER_BYPASS}" -eq 0 ] && grep -q "^${filepath}\$" devel/lint.ignore && return 0
     return 1
 }
```

**Review by frasertweedale — CHANGES_REQUESTED** [review]
Only one problem found.

File: devel/lint.sh (hunk 1 above)
Comment (avisiedo) [reply devel/lint.sh]: :+1:  @frasertweedale thanks a million mate, I change this.

**Comment by avisiedo** [comment]
Added some documentation at DEVOPS.md file.

File: devel/lint.sh (hunk 1 above)
Comment (avisiedo) [reply devel/lint.sh]: At the end I have used the expression `filepath="${filepath#./}"` that remove the indicated prefix if it matches.

Comment (frasertweedale) [reply devel/lint.sh]: Yep, that's a better solution.  Merging now, thanks!

**Pull request merged**